### PR TITLE
Rename DLContext to DLDevice

### PR DESCRIPTION
**Pull request by icemelon9**
This is the PR for the RFC #61 .

As discussed in the RFC, we will rename `DLContext` to `DLDevice` and provide `DLContext` as an alias via typedef.

cc: @tqchen @junrushao1994 @leofang @szha @kkraus14

### CODE REVIEW

File: include/dlpack/dlpack.h [hunk 1]
```diff
@@ -64,14 +64,16 @@ typedef enum {
...
  */
 typedef struct {
   /*! \brief The device type used in the device. */
   DLDeviceType device_type;
   /*! \brief The device index */
   int device_id;
-} DLContext;
+} DLDevice;
+
+typedef DLDevice DLContext;
```
Comment (tqchen) [inline include/dlpack/dlpack.h:76]: Please add a comment about the alias and indicating that we will consider remove it in the next release

**Review by tqchen — CHANGES_REQUESTED** [review]

File: include/dlpack/dlpack.h (hunk 1 above)
Comment (icemelon9) [reply include/dlpack/dlpack.h:79]: added the comment.

**Review by tqchen — APPROVED** [review]

**Review by junrushao1994 — APPROVED** [review]

**Comment by tqchen** [comment]
Thanks @icemelon9 ! This PR is being merged

**Comment by leofang** [comment]
Thanks!

File: include/dlpack/dlpack.h [hunk 2]
```diff
@@ -144,8 +149,8 @@ typedef struct {
    * \endcode
    */
   void* data;
-  /*! \brief The device context of the tensor */
-  DLContext ctx;
+  /*! \brief The device of the tensor */
+  DLDevice device;
```
Comment (leofang) [inline include/dlpack/dlpack.h:153]: @tqchen @icemelon9 Unfortunately this breaks existing C code. Could you please revert the field name to `ctx`?

Comment (tqchen) [reply include/dlpack/dlpack.h:153]: @leofang this is intentional to keep the code consistent. The existing codebase can either stay at previous version, or upgrade the code to move to the new version. Exchange won't be affected by the version being used

Comment (leofang) [reply include/dlpack/dlpack.h:153]: Ah OK, thanks. I failed to see this implication earlier.

**Pull request merged**